src/poller.ts

```diff
@@ -0,0 +1,97 @@
+import { DIPDeposit } from "./common";
+import {
+  Connection,
+  AccountChangeCallback,
+  PublicKey,
+  AccountInfo as solanaAccountInfo,
+  Context,
+  clusterApiUrl,
+} from "@solana/web3.js";
+import { AccountInfo, AccountLayout, u64 } from "@solana/spl-token";
+
+export class Poller {
+  cluster: string;
+  callback: (deposit: DIPDeposit) => void;
+  splToken: string;
+  premiumAsset: string;
+  expiration: number;
+  strike: number;
+  type: string;
+
+  constructor(cluster: string, callback: (deposit: DIPDeposit) => void) {
+    this.cluster = cluster;
+    this.callback = callback;
+    this.splToken = "BTC";
+    this.premiumAsset = "USDC";
+    this.expiration = Date.now();
+    this.strike = 25000;
+    this.type = "call";
+  }
+
+  subscribe(address: string): void {
+    console.log("Listening at:", address);
+    let previous_amount = 0;
+    const connection: Connection = new Connection(clusterApiUrl("devnet"));
+    const callback: AccountChangeCallback = (
+      accountInfo: solanaAccountInfo<Buffer>,
+      _context: Context
+    ) => {
+      // @ts-ignore
+      let new_amount = parseTokenAccount(accountInfo.data).amount.toNumber();
```

Comment:
checking if matches any other DIP in existence?

Reply:
This is the cumulative amount.